evennia/accounts/bots.py

```diff
@@ -576,3 +559,186 @@ def execute_cmd(
+                if not channel:
+                    raise RuntimeError(f"Evennia Channel {channel_name} not found.")
+                channel = channel[0]
+                self.ndb.ev_channels[channel_name] = channel
+
+    def start(self):
+        """
+        Tell the Discord protocol to connect.
+
+        """
+        if not _DISCORD_ENABLED:
+            self.delete()
+            return
+
+        if self.ndb.ev_channels:
+            for channel in self.ndb.ev_channels.values():
+                channel.connect(self)
+
+        elif channel_links := self.db.channels:
+            # this attribute contains a list of evennia<->discord links in the form of ("evennia_channel", "discord_chan_id")
+            # grab Evennia channels, cache and connect
+            channel_set = {evchan for evchan, dcid in channel_links}
+            self.ndb.ev_channels = {}
+            for channel_name in list(channel_set):
+                channel = search.search_channel(channel_name)
+                if not channel:
+                    raise RuntimeError(f"Evennia Channel {channel_name} not found.")
+                channel = channel[0]
+                self.ndb.ev_channels[channel_name] = channel
+                channel.connect(self)
+
+        # connect
+        global _SESSIONS
+        if not _SESSIONS:
+            from evennia.server.sessionhandler import SESSIONS as _SESSIONS
+        # these will be made available as properties on the protocol factory
+        configdict = {"uid": self.dbid}
+        _SESSIONS.start_bot_session(self.factory_path, configdict)
+
+    def at_pre_channel_msg(self, message, channel, senders=None, **kwargs):
+        """
+        Called by the Channel just before passing a message into `channel_msg`.
+
+        We overload this to force off the channel tag prefix.
+        """
+        kwargs["no_prefix"] = not self.db.tag_channel
+        return super().at_pre_channel_msg(message, channel, senders=senders, **kwargs)
+
+    def channel_msg(self, message, channel, senders=None, **kwargs):
+        """
+        Passes channel messages received on to discord
+
+        Args:
+            message (str): Incoming text from channel.
+            channel (Channel): The channel the message is being received from
+
+        Keyword Args:
+            senders (list or None): Object(s) sending the message
+
+        """
+        if kwargs.get("relayed"):
+            # don't relay our own relayed messages
+            return
+        if channel_list := self.db.channels:
+            # get all the discord channels connected to this evennia channel
+            channel_name = channel.name
+            for dc_chan in [dcid for evchan, dcid in channel_list if evchan == channel_name]:
+                # send outputfunc channel(msg, discord channel)
+                super().msg(channel=(strip_ansi(message.strip()), dc_chan))
+
+    def direct_msg(self, message, sender, **kwargs):
```

Comment:
Mark what the `**kwargs` are for too; in this case they are passed in into the relay. Are they unused? Are there any particular keywords we know of that can be used?